### PR TITLE
Bright Disputes - Milestone 1 delivery

### PR DESCRIPTION
**Pull request by bufta**

### CODE REVIEW

**Comment by TtomaS7** [comment]
Thanks for the delivery! 
Curators will take a look shortly!

**Comment by HCastano** [comment]
Thanks for the submission! I'll look through it these next couple of days 😄

File: deliveries/Bright_Disputes-Milestone_1.md
Comment (HCastano) [inline deliveries/Bright_Disputes-Milestone_1.md:16]:
```suggestion
```

File: deliveries/Bright_Disputes-Milestone_1.md
Comment (HCastano) [inline deliveries/Bright_Disputes-Milestone_1.md:4]:
```suggestion
```

**Review by HCastano — COMMENTED** [review]
Hey guys!

Here are some thoughts and observations from my first look.
- Why are you using a nightly from last year?
- Why are you using the a `2.x` series release of `cargo-contract`?
- The term "Jure" should be "Juror"
- You should have more documentation around all public items
- The autor email in the project `Cargo.toml` files should be in brackets
    - https://doc.rust-lang.org/cargo/reference/manifest.html#the-authors-field
- Why are your E2E tests in their own crate instead of being written as `ink_e2e` tests?
    - Not saying this is necessarily bad, but curious why you went with this approach
- Can you link to the source code for the Docker images published to Amazon ECR and used
  in the `deploy.sh` script?
- Nice to see that all your unit tests and E2E tests pass 👌

I still want to dig into the tests and see that all the basic functionality is covered,
as well as walk through the docs you provided manually testing the workflow with the UI.

Good work so far!

**Comment by mgralinski-bright** [comment]
> Hey guys!
> 
> Here are some thoughts and observations from my first look.
> 
> * Why are you using a nightly from last year?
> * Why are you using the a `2.x` series release of `cargo-contract`?
> * The term "Jure" should be "Juror"
> * You should have more documentation around all public items
> * The autor email in the project `Cargo.toml` files should be in brackets
>   
>   * https://doc.rust-lang.org/cargo/reference/manifest.html#the-authors-field
> * Why are your E2E tests in their own crate instead of being written as `ink_e2e` tests?
>   
>   * Not saying this is necessarily bad, but curious why you went with this approach
> * Can you link to the source code for the Docker images published to Amazon ECR and used
>   in the `deploy.sh` script?
> * Nice to see that all your unit tests and E2E tests pass 👌
> 
> I still want to dig into the tests and see that all the basic functionality is covered, as well as walk through the docs you provided manually testing the workflow with the UI.
> 
> Good work so far!

Hi @HCastano , 

thanks for sharing your thoughts, let me try to answer on them:

1. We have picked the older version of the rust, because we wanted to be consistent with the aleph toolchain. In the first milestone we only introduce a smart-contract, but in next we are going to extend aleph libraries with our code. We just wanted to avoid any further complications.
2. Newer version of cargo-contract requires a newer version of rust.
3. E2E test are using ink-wrapper which coverts smart-contract to the type-safe code. We are going to use this code in the next milestone for building a CLI. By doing it this way, it assures us that the CLI calls will work fine.
4. We don't have access to the resources published on the Amazon ERC. As we mentioned in the [documentation](https://github.com/bright/bright-disputes/blob/main/doc/README.md), deploy script was based on the `shielder` dApp. In the next milestones we will need to provide our own resources, because we are going to extend `liminal-ark-relations`.

**Comment by HCastano** [comment]
> Hi @HCastano ,
> 
> thanks for sharing your thoughts, let me try to answer on them:
> 
>     1. We have picked the older version of the rust, because we wanted to be consistent with the aleph toolchain. In the first milestone we only introduce a smart-contract, but in next we are going to extend aleph libraries with our code. We just wanted to avoid any further complications.

Can you link to where in the Aleph docs it says to use such an old nightly? From my reading they mention using a [recent nightly](https://docs.alephzero.org/aleph-zero/build/aleph-zero-smart-contracts-basics/installing-required-tools) and/or a [stable toolchain](https://docs.alephzero.org/aleph-zero/security-course-by-kudelski-security/ink-developers-security-guideline).


>     3. E2E test are using ink-wrapper which coverts smart-contract to the type-safe code. We are going to use this code in the next milestone for building a CLI. By doing it this way, it assures us that the CLI calls will work fine.

That makes sense. Any plans to convert them to `ink_e2e` tests in the future after the CLI has been tested?
 
>     4. We don't have access to the resources published on the Amazon ERC. As we mentioned in the [documentation](https://github.com/bright/bright-disputes/blob/main/doc/README.md), deploy script was based on the `shielder` dApp. In the next milestones we will need to provide our own resources, because we are going to extend `liminal-ark-relations`.

So how do you know what's inside each image in terms of packages and versions? And who is publishing these images? I'd guess the CC team, but it's unclear from just the ERC page.

**Comment by mgralinski-bright** [comment]
@HCastano 
 
> Can you link to where in the Aleph docs it says to use such an old nightly? From my reading they mention using a [recent nightly](https://docs.alephzero.org/aleph-zero/build/aleph-zero-smart-contracts-basics/installing-required-tools) and/or a [stable toolchain](https://docs.alephzero.org/aleph-zero/security-course-by-kudelski-security/ink-developers-security-guideline).
> 

You were right, we can use a stable version but 1.69 not the latest. I couldn't build it with the latest version, because I faced some issues (`Unknown opcode 192` on post processing). I think I picked the old toolchain because I saw [shielder](https://github.com/Cardinal-Cryptography/zk-apps/blob/main/rust-toolchain) is using it. Anyway, by switching to Rust 1.69, we can use cargo-contract 3.0.1 instead 2.x. I have made new commit where I have also updated the ink-wrapper to version 0.5.0.

> 
> That makes sense. Any plans to convert them to `ink_e2e` tests in the future after the CLI has been tested?
> 

> > ```
> > 4. We don't have access to the resources published on the Amazon ERC. As we mentioned in the [documentation](https://github.com/bright/bright-disputes/blob/main/doc/README.md), deploy script was based on the `shielder` dApp. In the next milestones we will need to provide our own resources, because we are going to extend `liminal-ark-relations`.
> > ```
> 
> So how do you know what's inside each image in terms of packages and versions? And who is publishing these images? I'd guess the CC team, but it's unclear from just the ERC page.

**Comment by mgralinski-bright** [comment]
@HCastano 

> Can you link to where in the Aleph docs it says to use such an old nightly? From my reading they mention using a [recent nightly](https://docs.alephzero.org/aleph-zero/build/aleph-zero-smart-contracts-basics/installing-required-tools) and/or a [stable toolchain](https://docs.alephzero.org/aleph-zero/security-course-by-kudelski-security/ink-developers-security-guideline).
> 

You were right, we can use a stable version but 1.69 not the latest. I couldn't build it with the latest version, because I faced some issues (`Unknown opcode 192` on post processing). I think I picked the old toolchain because I saw [shielder](https://github.com/Cardinal-Cryptography/zk-apps/blob/main/rust-toolchain) is using it. Anyway, by switching to Rust 1.69, we can use cargo-contract 3.0.1 instead 2.x. I have made new commit where I have also updated the ink-wrapper to version 0.5.0.

> 
> That makes sense. Any plans to convert them to `ink_e2e` tests in the future after the CLI has been tested?
> 

We haven't plan to use `ink_e2e` for E2E tests, because in my opinion they are just a duplication of testa created with the `ink-wrapper`. However, we are open for the discussion. 

> So how do you know what's inside each image in terms of packages and versions? And who is publishing these images? I'd guess the CC team, but it's unclear from just the ERC page.

Regarding this, to be honest I didn't expect this is going to be a problem. I assumed that we could reuse those images for the `shielder` dApp, and in the next milestone provide our own resources. Will you agree on this? If you have any objections, please propose a solution and we will apply it.

**Comment by HCastano** [comment]
@mgralinski-bright 

> We haven't plan to use ink_e2e for E2E tests, because in my opinion they are just a duplication of testa created with the ink-wrapper. However, we are open for the discussion.

They'd reduce the boilerhead associated with your current testing setup. They're also nice since they live alongside the contract. There is room for both types of tests imo though.

> Regarding this, to be honest I didn't expect this is going to be a problem. I assumed that we could reuse those images for the shielder dApp, and in the next milestone provide our own resources. Will you agree on this? If you have any objections, please propose a solution and we will apply it.

It's not a _problem_ per say, but I would be wary of using images (or tools in general) that you don't have a good grasp on. What happens if something breaks and you need to debug? It's gonna be hard if you don't know what's inside. What happens if the owner pulls the image from the registry?

**Comment by mgralinski-bright** [comment]
@HCastano 

following your advice, we provide our own docker images (stored on the dockerhub) which are used by the script. Please download the latest changes from our repository, we also improved other suggestions. When we can expect the evaluation document?

**Review by HCastano — APPROVED** [review]
Re: Docker: There's still no indication of where the source for the Dockefiles you provided is. Be it yours or third party having access (or a link) to the source is important imo. For the new Docker images, how much do they differ from the CC provided ones? Anyways, this is something that can be improved in the future.

Thanks for the submission, LGTM!

**Comment by TtomaS7** [comment]
@DoubleOTheven please review the Delivery!

**Comment by bufta** [comment]
Hi @TtomaS7, I would just like to check in to ask about the evaluation - when could we expect the second review? Next week our team goes on vacation, we'll be out of office for 2 weeks and it would be difficult for us to respond to any comments then. We are still available and more than ready to answer any feedback issues this week though.

**Comment by TtomaS7** [comment]
Hello, 
Thank you for letting us know! I will ping curators, thanks

**Comment by DoubleOTheven** [comment]
Just a heads up: I'm getting an error on my Mac when running the Docker build. 

```sh
 => ERROR [build  5/14] RUN dpkg -i libssl1.1_1.1.1f-1ubuntu2_amd64.deb                                                                                                                   0.3s
------
 > [build  5/14] RUN dpkg -i libssl1.1_1.1.1f-1ubuntu2_amd64.deb:
#0 0.293 dpkg: error processing archive libssl1.1_1.1.1f-1ubuntu2_amd64.deb (--install):
#0 0.293  package architecture (amd64) does not match system (arm64)
#0 0.297 Errors were encountered while processing:
#0 0.297  libssl1.1_1.1.1f-1ubuntu2_amd64.deb
------
Dockerfile:14
--------------------
  12 |
  13 |     RUN wget http://archive.ubuntu.com/ubuntu/pool/main/o/openssl/libssl1.1_1.1.1f-1ubuntu2_amd64.deb
  14 | >>> RUN dpkg -i libssl1.1_1.1.1f-1ubuntu2_amd64.deb
  15 |
  16 |     RUN curl https://sh.rustup.rs -sSf | sh -s -- -y
--------------------
ERROR: failed to solve: process "/bin/sh -c dpkg -i libssl1.1_1.1.1f-1ubuntu2_amd64.deb" did not complete successfully: exit code: 1
```

**Comment by TtomaS7** [comment]
Hey, Bright Disputes team!
Congratulations on your delivery! I will share the form for the payment details on this week.

Thank you for cooperation!

**Comment by bufta** [comment]
Hi, I would just like to check if everything's alright and ask about the channel on which we should expect the form for the payment. Thank you!

**Comment by TtomaS7** [comment]
Hello, 
Yes, we're still waiting for the Election of curation team Proposal approval. 
I will share the payment form and the instructions, once it’s approved.
Thank you.

**Comment by bufta** [comment]
Hi @TtomaS7 as we're getting close to the delivery of our second milestone, I'd like to check in to see if there's any update or timeline regarding the payment for our first milestone? Should we wait with the second delivery until our first milestone is settled?

**Comment by TtomaS7** [comment]
Hey team!

Here's the form for payment details: https://forms.gle/BUCKfygkiEvpdYna6.
Once you've filled it out, please let me know and be sure to specify what it pertains to; you can include the link to this PR delivery! Further details can be provided under the section: "Please describe which Milestone you have completed and attach the link to the Milestone delivery."

Should you have any questions, don't hesitate to ask. Thank you so much!

Important Note: PLEASE VERIFY YOUR ADDRESS, WHERE YOU WILL RECEIVE THE FUNDS, BEFORE FILLING OUT THE FORM!
For guidance on how to set up identity, refer to [this guide](https://support.polkadot.network/support/solutions/articles/65000181981-how-to-set-and-clear-an-identity).

**Comment by bufta** [comment]
Thank you @TtomaS7! I've just filled the form 👍

**Comment by TtomaS7** [comment]
The payment sent!

https://polkadot.subsquare.io/treasury/child-bounties/19_459
You can claim it in 1 day, thank you!

**Pull request merged**
